http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/NettyUtils.java

```diff
@@ -40,6 +40,14 @@
      */
     public static final SucceededFuture<?> SUCCEEDED_FUTURE = new SucceededFuture<>(null, null);
 
+    public static final String CLOSED_CHANNEL_MESSAGE = "The channel was closed. This may have been done by the client (e.g. "
+                                                        + "because the request was aborted), " +
+                                                        "by the service (e.g. because there was a handshake error[use -Djavax"
+                                                        + ".net.debug=ssl to enable SSL logs], the request "
```

Comment:
Is there any way we can detect that it was a handshake error so that they don't have to enable SSL logs? I wouldn't want every customer having to enable SSL logs to see if it was a handshake error just a plain-ol'-closed-channel.

Reply:
No, unfortunately, this kind of errors are bubbled up as `ClosedChannelException` with no details at all. When I was troubleshooting the failing test, it took me while to figure out it was related to handshake.

Reply:
I wish we could link to docs that help customers debug these errors. Maybe something we can talk to @Carey-AWS about. In the meantime, it's really hard to read this with the parens-in-parens for the "use -D...". Can we exclude that? The other potential causes we list don't say how to diagnose them.

Reply:
+1. Yeah, a guide on how to troubleshoot closed channel exception would be super helpful!

Reply:
Thanks for calling this out, @millems. I've added it to the backlog.